prow/pod-utils/decorate/podspec.go

```diff
@@ -172,6 +168,167 @@ func ProwJobToPod(pj kube.ProwJob, buildID string) (*v1.Pod, error) {
+}
+
+// sshVolume converts a secret holding ssh keys into the corresponding volume and mount.
+//
+// This is used by MakeCloner to attach the mount to the clonerefs container.
+func sshVolume(secret string) (kube.Volume, kube.VolumeMount) {
+	var sshKeyMode int32 = 0400 // this is octal, so symbolic ref is `u+r`
+	name := strings.Join([]string{"ssh-keys", secret}, "-")
+	mountPath := path.Join("/secrets/ssh", secret)
+	v := kube.Volume{
+		Name: name,
+		VolumeSource: kube.VolumeSource{
+			Secret: &kube.SecretSource{
+				SecretName:  secret,
+				DefaultMode: &sshKeyMode,
+			},
+		},
+	}
+
+	vm := kube.VolumeMount{
+		Name:      name,
+		MountPath: mountPath,
+		ReadOnly:  true,
+	}
+
+	return v, vm
+}
+
+// cookiefileVolumes converts a secret holding cookies into the corresponding volume and mount.
+//
+// This is used by MakeCloner to attach the mount to the clonerefs container.
+// The returned string value is the path to the cookiefile for use with --cookiefile.
+func cookiefileVolume(secret string) (kube.Volume, kube.VolumeMount, string) {
+	// my-cookie/.gitcookies => find my-cookie secret at /secrets/cookiefile/.gitcookies
+	// my-cookie => find my-cookie secret at /secrets/cookiefile/my-cookie
+	parts := strings.SplitN(secret, "/", 2)
+	cookieSecret := parts[0]
+	var base string
+	if len(parts) == 1 {
+		base = parts[0]
+	} else {
+		base = parts[1]
+	}
+	var cookiefileMode int32 = 0400 // u+r
+	vol := kube.Volume{
+		Name: "cookiefile",
+		VolumeSource: kube.VolumeSource{
+			Secret: &kube.SecretSource{
+				SecretName:  cookieSecret,
+				DefaultMode: &cookiefileMode,
+			},
+		},
+	}
+	mount := kube.VolumeMount{
+		Name:      vol.Name,
+		MountPath: "/secrets/cookiefile", // append base to flag
+		ReadOnly:  true,
+	}
+	return vol, mount, path.Join(mount.MountPath, base)
+}
+
+// MakeCloner converts a prow job, code and log volume mounts and into a clonerefs container.
+//
+// Returns the clonerefs container (nil if no cloning) plus additional volumes required by this container.
+//
+// The clonerefs container will check out repositories specified by the prow job to the code mount.
+//
+// The cloneref logging will be written to the log mount, which is
+// typically consumed by the initupload container.
+//
+// If the job defines ssh keys and/or cookiefiles then these secrets will be mounted into the container.
+// The volumes required for these secrets will included in the returned volume list.
+func MakeCloner(pj kube.ProwJob, codeMount, logMount kube.VolumeMount) (*kube.Container, []kube.Volume, error) {
+	if pj.Spec.DecorationConfig == nil {
+		pj.Spec.DecorationConfig = &kube.DecorationConfig{}
+	}
+	if pj.Spec.DecorationConfig.SkipCloning {
+		return nil, nil, nil
+	}
+	var cloneVolumes []kube.Volume
+	var refs []*kube.Refs
+	if pj.Spec.Refs != nil {
+		refs = append(refs, pj.Spec.Refs)
+	}
+	refs = append(refs, pj.Spec.ExtraRefs...)
+	if len(refs) == 0 {
```

Comment:
Same comment re: `periodic` jobs, why not conditionally call this function for the appropriate types of jobs?